internal/plugins/helm/init.go

```diff
@@ -0,0 +1,157 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package helm
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	gencrd "github.com/operator-framework/operator-sdk/internal/generate/crd"
+	sdkscaffold "github.com/operator-framework/operator-sdk/internal/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/helm"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/input"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+)
+
+type initPlugin struct {
+	config *config.Config
+
+	// For help text.
+	commandName string
+
+	// Helm flags
+	APIVersion       string
+	Kind             string
+	CRDVersion       string
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
+}
+
+var (
+	_ plugin.Init = &initPlugin{}
+)
+
+func (p *initPlugin) UpdateContext(ctx *plugin.Context) {
+	ctx.Description = `Initialize a Helm project with the Helm Chart package directories.`
+	p.commandName = ctx.CommandName
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	const (
+		apiVersion       = "api-version"
+		kind             = "kind"
+		crdVersion       = "crd-version"
+		helmChart        = "helm-chart"
+		helmChartVersion = "helm-chart-version"
+		helmChartRepo    = "helm-chart-repo"
+	)
+
+	fs.StringVar(&p.APIVersion, apiVersion, "",
+		"Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)")
+	fs.StringVar(&p.Kind, kind, "", "Kubernetes resource Kind name. (e.g AppService)")
+	fs.StringVar(&p.CRDVersion, crdVersion, gencrd.DefaultCRDVersion,
+		"CRD version to generate")
+	fs.StringVar(&p.HelmChartRef, helmChart, "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path).")
+	fs.StringVar(&p.HelmChartVersion, helmChartVersion, "",
+		"Specific version of the helm chart (default is latest version)")
+	fs.StringVar(&p.HelmChartRepo, helmChartRepo, "",
+		"Chart repository URL for the requested helm chart")
+
+}
+
+func (p *initPlugin) InjectConfig(c *config.Config) {
+	c.Layout = plugin.KeyFor(Plugin{})
+	p.config = c
+}
+
+func (p *initPlugin) Run() error {
+	if err := p.Validate(); err != nil {
+		return err
+	}
+	if err := p.Scaffold(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *initPlugin) Validate() error {
+	// Check if the project name is a valid namespace according to k8s
+	dir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("error to get the current path: %v", err)
+	}
+	projectName := filepath.Base(dir)
+
+	if err := validation.IsDNS1123Label(strings.ToLower(projectName)); err != nil {
+		return fmt.Errorf("project name (%s) is invalid: %v", projectName, err)
+	}
+
+	if len(p.HelmChartRef) == 0 {
```

Comment:
Nit: these empty string comparisons would be a lot clearer if you did
```suggestion
	if p.HelmChartRef == "" {
```
Obviously both comparisons are valid, but the reason the string comparison exists is to clearly identify the type of a field/variable.

Reply:
@estroz we should NOT only check == "" because it is the flag value and can be a string such as "      " which would be invalid in the same way. E.g if we call the command with ``--helm-chart="       "`` However, it is missing the trim. 👍

Reply:
They'd fail in the same way if the string contained spaces...

Reply:
It was missing do a trim first which is done now. See;

https://github.com/operator-framework/operator-sdk/pull/3239/files#diff-7bb5b0d0fc8ac97531d01469bfaa65a1R61